### PR TITLE
Return chat component with empty text instead of throwing exception.

### LINKED ISSUE #3328: Console Error - InventoryClickEvent
I have four different plugins giving me the same exact error, i have no means to reproduce the errors and it doesn't seem to break their behaviors in any way.

> Could not pass event InventoryClickEvent to "plugin name"
> java.lang.IllegalStateException: Title not set

This is one full log example: https://pastebin.com/raw/Fj5rLGgp

This server is running Paper version git-Paper-264 (MC: 1.15.2)
Java 8

### PR DESCRIPTION
**Pull request by CDFN**
Fixes #3328

### CODE REVIEW

File: Spigot-API-Patches/0218-Return-chat-component-with-empty-text-instead-of-thr.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+--- a/src/main/java/org/bukkit/inventory/InventoryView.java
++++ b/src/main/java/org/bukkit/inventory/InventoryView.java
+@@ -449,7 +449,7 @@ public abstract class InventoryView {
+     /**
+      * Get the title of this inventory window.
+      *
+-     * @return The title.
++     * @return The title or empty string when title is null.
+      */
+     @NotNull
```
Comment (Proximyst) [inline Spigot-API-Patches/0218-Return-chat-component-with-empty-text-instead-of-thr.patch:19]: This isn't not null if the title can be null, eh?

EDIT: Misread & cleared in DMs :p

File: Spigot-Server-Patches/0542-Return-chat-component-with-empty-text-instead-of-thr.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
++++ b/src/main/java/net/minecraft/server/Container.java
+@@ -45,7 +45,12 @@ public abstract class Container {
+     }
+     private IChatBaseComponent title;
+     public final IChatBaseComponent getTitle() {
+-        Preconditions.checkState(this.title != null, "Title not set");
++        // Paper start - return chat component with empty text instead of throwing error
++        // Preconditions.checkState(this.title != null, "Title not set");
++        if(this.title == null){
++            return IChatBaseComponent.ChatSerializer.jsonToComponent("\"text\":\"\"");
```
Comment (Proximyst) [inline Spigot-Server-Patches/0542-Return-chat-component-with-empty-text-instead-of-thr.patch]: Does this even work? Would it not require being a JSON object?

**Review by Proximyst — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0542-Return-chat-component-with-empty-text-instead-of-thr.patch (hunk 2 above)
Comment (CDFN) [reply Spigot-Server-Patches/0542-Return-chat-component-with-empty-text-instead-of-thr.patch]: Good point, thanks 👍

File: Spigot-Server-Patches/0542-Return-chat-component-with-empty-text-instead-of-thr.patch [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
++++ b/src/main/java/net/minecraft/server/Container.java
+@@ -45,7 +45,12 @@ public abstract class Container {
+     }
+     private IChatBaseComponent title;
+     public final IChatBaseComponent getTitle() {
+-        Preconditions.checkState(this.title != null, "Title not set");
++        // Paper start - return chat component with empty text instead of throwing error
++        // Preconditions.checkState(this.title != null, "Title not set");
++        if(this.title == null){
++            return IChatBaseComponent.ChatSerializer.jsonToComponent("{\"text\":\"\"}");
```
Comment (aikar) [inline Spigot-Server-Patches/0542-Return-chat-component-with-empty-text-instead-of-thr.patch]: new ChatComponentText("")

**Review by aikar — CHANGES_REQUESTED** [review]

File: Spigot-API-Patches/0218-Return-chat-component-with-empty-text-instead-of-thr.patch [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+diff --git a/src/main/java/org/bukkit/inventory/InventoryView.java b/src/main/java/org/bukkit/inventory/InventoryView.java
+index 14346d83bc99581b18e53d19af03708c0bf22cf7..98d8242122dff5811e689143cb26aaadd4b1d064 100644
+--- a/src/main/java/org/bukkit/inventory/InventoryView.java
++++ b/src/main/java/org/bukkit/inventory/InventoryView.java
+@@ -449,7 +449,7 @@ public abstract class InventoryView {
+     /**
+      * Get the title of this inventory window.
+      *
+-     * @return The title.
++     * @return The title or empty string when title is null.
```
Comment (chickeneer) [inline Spigot-API-Patches/0218-Return-chat-component-with-empty-text-instead-of-thr.patch:17]: This should get a `// Paper` tag on it. Just at the end of the line.

**Review by chickeneer — CHANGES_REQUESTED** [review]

File: Spigot-API-Patches/0218-Return-chat-component-with-empty-text-instead-of-thr.patch (hunk 4 above)
Comment (Proximyst) [reply Spigot-API-Patches/0218-Return-chat-component-with-empty-text-instead-of-thr.patch:17]: The javadoc is HTML, so:
```suggestion
+     * @return The title or empty string when title is {@code null}. <!-- Paper -->
```

**Review by Proximyst — APPROVED** [review]

**Review by electronicboy — APPROVED** [review]

**Comment by MiniDigger** [comment]
https://github.com/PaperMC/Paper/pull/5070

thanks

**Pull request closed without merging**